Code/Core/Env/glibc_compat_stat.h

```diff
@@ -0,0 +1,39 @@
+// glibc_compat_stat
+//
+// Use stat,lstat,fstat implementations from older versions of glibc for better Linux distro compatibility
+//------------------------------------------------------------------------------
+#pragma once
+
+// Includes
+//------------------------------------------------------------------------------
+#include <sys/stat.h>
+
+// In glibc 2.33 implementation of various stat functions was changed - usage of proxy functions (__xstat, etc.) was removed.
+// As a result code built with glibc 2.33 headers links to stat@GLIBC_2.33 instead of __xstat@GLIBC_2.2.5.
+// But in this case we can't use the __asm__(".symver") trick to use old versions of those functions,
+// because there are no compatible versions - those proxy functions have a different signature.
+// Thus we must use a more involved hack to force usage of old versions to avoid depending on a new glibc version.
+
+#if defined( __GLIBC__ ) && ( __GLIBC__ * 1000 + __GLIBC_MINOR__ ) >= 2033
+
+    extern "C" int __xstat ( int ver, const char * path, struct stat * buf );
+    extern "C" __inline int stat ( const char * path, struct stat * buf )
+    {
+        return __xstat( 0 /*_STAT_VER*/, path, buf );
+    }
```

Comment:
This is more or less corresponds to what glibc does: https://sourceware.org/git/?p=glibc.git;a=blob;f=io/sys/stat.h;h=ce014d03a588ef5e2064b25594c69b9746c37976;hb=3de512be7ea6053255afed6154db9ee31d4e557a#l452